tools/amGithubNotifier/main.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-github/v26/github"
+	"github.com/prometheus/alertmanager/notify/webhook"
+	"github.com/prometheus/alertmanager/template"
+	"golang.org/x/oauth2"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type ghWebhookReceiverConfig struct {
+	authFile     string
+	defaultOwner string
+	defaultRepo  string
+	portNo       string
+	dryRun       bool
+}
+
+type ghWebhookReceiver struct {
+	ghClient *github.Client
+	cfg      ghWebhookReceiverConfig
+}
+
+type ghWebhookHandler struct {
+	client *ghWebhookReceiver
+}
+
+func main() {
+	/*
+		Example `alerts.rules.yml`:
+		```
+		groups:
+		- name: groupname
+		  rules:
+		  - alert: alertname
+		    expr: up == 0
+		    for: 2m
+		    labels:
+		      severity: average
+		      prNum: '{{ $labels.prNum }}'
+		      owner: prometheus (optional)
+		      repo: prometheus (optional)
+		    annotations:
+		      description: 'description of the alert'
+		```
+	*/
+	cfg := ghWebhookReceiverConfig{}
+
+	app := kingpin.New(filepath.Base(os.Args[0]), `alertmanager github webhook receiver
+	Example: ./amGithubNotifier --org=prometheus --repo=prometheus --port=8080`)
+	app.Flag("authfile", "path to github oauth token file").Default("/etc/github/oauth").StringVar(&cfg.authFile)
+	app.Flag("org", "default org/owner").Required().StringVar(&cfg.defaultOwner)
+	app.Flag("repo", "default repo").Required().StringVar(&cfg.defaultRepo)
```

Comment:
This is a bit confusing. Default for me is some default when not set and here is it always set from the cli arg.
the same for `org`